src/sentry/api/endpoints/project_details.py

```diff
@@ -223,6 +227,10 @@ def put(self, request, project):
             project.name = result['name']
             changed = True
 
+        if result.get('platform'):
```

Comment:
Matt had brought this up and I talked to Eric about it and we decided on leaving this flexible unless someone had major objections on it